catalog/spatial/registry/registry-report-viewer/src/main/java/org.codice.ddf.registry.report.viewer/RegistryReportViewer.java

```diff
@@ -0,0 +1,486 @@
+                    .build();
+        }
+
+        RegistryPackageType registryPackage;
+        try {
+            registryPackage = federationAdminService.getRegistryObjectByRegistryId(registryId,
+                    sourceIds);
+        } catch (FederationAdminException e) {
+            String message = "Error getting registry package.";
+            LOGGER.error("{} For metacard id: '{}', optional sources: {}",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
+
+        if (registryPackage == null) {
+            String message = "No registry package was found.";
+            LOGGER.error("{} For metacard id: '{}', optional source ids: {}.",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(message)
+                    .build();
+        }
+
+        Map<String, Object> registryMap = buildRegistryMap(registryPackage);
+
+        try {
+            Template template = handlebars.compile("report");
+            html = template.apply(registryMap);
+        } catch (IOException e) {
+            LOGGER.error("Error compiling and applying report template.");
+        }
+
+        return Response.ok(html)
+                .build();
+    }
+
+    protected Map<String, Object> buildRegistryMap(RegistryPackageType registryPackage) {
+        Map<String, Object> registryMap = new HashMap<>();
+
+        Map<String, Object> extrinsicInfo = getExtrinsicInfo(registryPackage);
+        Map<String, Object> generalMap = (Map<String, Object>) extrinsicInfo.get(GENERAL);
+        Map<String, Object> collectionMap = (Map<String, Object>) extrinsicInfo.get(COLLECTIONS);
+
+        registryMap.put(NAME, generalMap.get(REGISTRY_NAME));
+        generalMap.remove(REGISTRY_NAME);
+
+        registryMap.put(GENERAL, generalMap);
+        registryMap.put(COLLECTIONS, collectionMap);
+
+        Map<String, Object> serviceMap = getServiceInfo(registryPackage.getRegistryObjectList());
+        registryMap.put(SERVICES, serviceMap);
+
+        Map<String, Object> organizationMap = getOrganizationInfo(registryPackage);
+        registryMap.put(ORGANIZATIONS, organizationMap);
+
+        Map<String, Object> contactMap = getContactInfo(registryPackage);
+        registryMap.put(CONTACTS, contactMap);
+
+        return registryMap;
+    }
+
+    private Map<String, Object> getServiceInfo(RegistryObjectListType registryObjectListType) {
+        Map<String, Object> serviceInfo = new HashMap<>();
+        for (JAXBElement id : registryObjectListType.getIdentifiable()) {
+            RegistryObjectType registryObjectType = (RegistryObjectType) id.getValue();
+            if (registryObjectType instanceof ServiceType
```

Comment:
You could reduce the cyclomatic complexity of this loop by inverting the condition here:
```java
if (!(registryObjectType instanceof ServiceType) || !RegistryConstants.REGISTRY_SERVICE_OBJECT_TYPE.equals(registryObjectType.getObjectType()) {
    continue;
}
...
```
It's really a tossup. The reduction in complexity is a plus but it doesn't buy you much and does look a little odder.